src/error.jl

```diff
@@ -0,0 +1,62 @@
+"""
+    H5Error
+
+An error thrown by libhdf5.
+"""
+mutable struct H5Error <: Exception
+    msg::String
+    id::hid_t
+end
+
+
+Base.cconvert(::Type{hid_t}, err::H5Error) = err
+Base.unsafe_convert(::Type{hid_t}, err::H5Error) = err.id
+
+function Base.close(err::H5Error)
+    if err.id != -1 && isvalid(err)
+        h5e_close_stack(err)
+        err.id = -1
+    end
+    return nothing
+end
+Base.isvalid(err::H5Error) = err.id != -1 && h5i_is_valid(err)
+
+Base.length(err::H5Error) = h5e_get_num(err)
+Base.isempty(err::H5Error) = length(err) == 0
+
+function H5Error(msg::AbstractString)
+    id = h5e_get_current_stack()
+    err = H5Error(msg, id)
+    finalizer(close, err)
+    return err
+end
+
+const SHORT_ERROR = Ref(true)
+
+function Base.showerror(io::IO, err::H5Error)
+    n_total = length(err)
+    print(io, "H5Error: ", err.msg)
+    print(io, "\lLibrary Stacktrace:")
```

Comment:
```suggestion
    print(io, "\nLibrary Stacktrace:")
```
had a typo in my diff

Or perhaps call this 
```suggestion
    print(io, "\nlibhdf5 Stacktrace:")
```
or some variation of that, to make it clear this is coming from the libhdf5 shared library and not HDF5.jl ?